### PR TITLE
Fix checkbox runtime series16 snap (infra)

### PR DESCRIPTION
**Pull request by Hook25**
## Description

Currently the series16 snap fails to build (see: https://github.com/canonical/checkbox/actions/runs/6154735028).

This happens because the setuptools (and pip) version in the snap when running the start of the build-override section are very old. They were upgraded after the `snapcraft-build` step and this did not cause any issue because that step would not try to install the module as we had removed the setup.py file (and these old versions of pip/setuptools would just ignore the command). Since the file was re-introduced the build now fails because it tries to install using the backend declared in `pyproject.toml` (`setuptools.build_meta`, as you can see from the error) that is not present in this old version of setuptools.

This PR fixes the problem by preparing  the environment so that  the `snapcraft-build` is able to install the modules in both parts. 

## Resolved issues

(see: https://github.com/canonical/checkbox/actions/runs/6154735028).

## Documentation

This removes most hacks in the recipe and updates the comments in the overwritten build to be more significant

## Tests

Run the following to get a snap: (or refer to the snap building guide)
```
$ cd checkbox-core-snap
$ ./prepare.sh series16
$ cd series16
$ snapcraft remote-build 
```

Use the following config to test the snap via metabox:
```
configuration = {
    "local": {
        "origin": "classic-snap",
        "checkbox_core_snap": {
            "uri": "~/checkbox16.snap"
        },
        "checkbox_snap": {"risk": "edge"},
        "releases": ["xenial"],
    },
    "remote": {
        "origin": "classic-snap",
        "checkbox_core_snap": {
            "uri": "~/checkbox16.snap"
        },
        "checkbox_snap": {"risk": "edge"},
        "releases": ["xenial"],
    },
    "service": {
        "origin": "classic-snap",
        "checkbox_core_snap": {
            "uri": "~/checkbox16.snap"
        },
        "checkbox_snap": {"risk": "edge"},
        "releases": ["xenial"],
    },
}
```

### CODE REVIEW

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 1]
```diff
@@ -147,13 +147,15 @@ parts:
       - python3-systemd
       - python3-setuptools
       - libbluetooth3
-    python-packages:
-      - pynmea2
-      - pybluez
-    after: [acpi-tools]
+        #python-packages:
+        #- pynmea2
```
Comment (kissiel) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:151]: delete, don't comment out

**Review by kissiel — COMMENTED** [review]
One request below.

**Review by kissiel — APPROVED** [review]
And as a bonus: Negative diff!
GET IN! +1

**Comment by Hook25** [comment]
Marking as draft as the produced snap doesn't pass metabix validation

**Comment by Hook25** [comment]
Remote Tests:
```
17:19:33 | INFO     Starting scenario (1/35): [remote][(xenial, xenial)] basic.return_codes.ReturnCodeIsOneOnFail
17:19:43 | SUCCESS  [remote][(xenial, xenial)] basic.return_codes.ReturnCodeIsOneOnFail scenario has passed.
17:19:55 | INFO     Starting scenario (2/35): [remote][(xenial, xenial)] config.daemon.DaemonNormalUserSetInLauncherNoConfig
17:20:10 | SUCCESS  [remote][(xenial, xenial)] config.daemon.DaemonNormalUserSetInLauncherNoConfig scenario has passed.
17:20:21 | INFO     Starting scenario (3/35): [remote][(xenial, xenial)] config.daemon.DaemonNormalUserSetInConfig
17:20:36 | SUCCESS  [remote][(xenial, xenial)] config.daemon.DaemonNormalUserSetInConfig scenario has passed.
17:20:51 | INFO     Starting scenario (4/35): [remote][(xenial, xenial)] config.daemon.DaemonNormalUserOverwittenByLauncher
17:21:06 | SUCCESS  [remote][(xenial, xenial)] config.daemon.DaemonNormalUserOverwittenByLauncher scenario has passed.
17:21:19 | INFO     Starting scenario (5/35): [remote][(xenial, xenial)] config.daemon.DaemonNormalUserGuessed
17:21:37 | SUCCESS  [remote][(xenial, xenial)] config.daemon.DaemonNormalUserGuessed scenario has passed.
17:21:48 | INFO     Starting scenario (6/35): [remote][(xenial, xenial)] config.daemon.DaemonNormalUserDoesntExist
17:21:55 | SUCCESS  [remote][(xenial, xenial)] config.daemon.DaemonNormalUserDoesntExist scenario has passed.
17:22:06 | INFO     Starting scenario (7/35): [remote][(xenial, xenial)] config.environment.CheckboxConfEnvvarCaseSensitive
17:22:23 | SUCCESS  [remote][(xenial, xenial)] config.environment.CheckboxConfEnvvarCaseSensitive scenario has passed.
17:22:34 | INFO     Starting scenario (8/35): [remote][(xenial, xenial)] config.environment.CheckboxConfXDG
17:22:51 | SUCCESS  [remote][(xenial, xenial)] config.environment.CheckboxConfXDG scenario has passed.
17:23:03 | INFO     Starting scenario (9/35): [remote][(xenial, xenial)] config.environment.CheckboxConfRemoteHome
17:23:20 | SUCCESS  [remote][(xenial, xenial)] config.environment.CheckboxConfRemoteHome scenario has passed.
17:23:31 | INFO     Starting scenario (10/35): [remote][(xenial, xenial)] config.environment.CheckboxConfSnap
17:23:48 | SUCCESS  [remote][(xenial, xenial)] config.environment.CheckboxConfSnap scenario has passed.
17:23:59 | INFO     Starting scenario (11/35): [remote][(xenial, xenial)] config.environment.CheckboxConfLauncher
17:24:15 | SUCCESS  [remote][(xenial, xenial)] config.environment.CheckboxConfLauncher scenario has passed.
17:24:26 | INFO     Starting scenario (12/35): [remote][(xenial, xenial)] config.environment.CheckboxConfLauncherPrecedence
17:24:43 | SUCCESS  [remote][(xenial, xenial)] config.environment.CheckboxConfLauncherPrecedence scenario has passed.
17:24:54 | INFO     Starting scenario (13/35): [remote][(xenial, xenial)] config.environment.CheckboxConfRemoteServiceResolutionOrder
17:25:12 | SUCCESS  [remote][(xenial, xenial)] config.environment.CheckboxConfRemoteServiceResolutionOrder scenario has passed.
17:25:23 | INFO     Starting scenario (14/35): [remote][(xenial, xenial)] config.exporters.ExporterSettingOptions
17:25:39 | SUCCESS  [remote][(xenial, xenial)] config.exporters.ExporterSettingOptions scenario has passed.
17:25:50 | INFO     Starting scenario (15/35): [remote][(xenial, xenial)] config.test_manifest.ManifestLauncherAuto
17:26:10 | SUCCESS  [remote][(xenial, xenial)] config.test_manifest.ManifestLauncherAuto scenario has passed.
17:26:21 | INFO     Starting scenario (16/35): [remote][(xenial, xenial)] config.test_manifest.ManifestLauncherManual
17:26:33 | SUCCESS  [remote][(xenial, xenial)] config.test_manifest.ManifestLauncherManual scenario has passed.
17:26:45 | INFO     Starting scenario (17/35): [remote][(xenial, xenial)] config.test_manifest.ManifestConfigCacheAuto
17:27:06 | SUCCESS  [remote][(xenial, xenial)] config.test_manifest.ManifestConfigCacheAuto scenario has passed.
17:27:17 | INFO     Starting scenario (18/35): [remote][(xenial, xenial)] config.test_manifest.ManifestConfigCacheManual
17:27:31 | SUCCESS  [remote][(xenial, xenial)] config.test_manifest.ManifestConfigCacheManual scenario has passed.
17:27:44 | INFO     Starting scenario (19/35): [remote][(xenial, xenial)] config.test_manifest.ManifestConfigPrecedenceAuto
17:28:05 | SUCCESS  [remote][(xenial, xenial)] config.test_manifest.ManifestConfigPrecedenceAuto scenario has passed.
17:28:16 | INFO     Starting scenario (20/35): [remote][(xenial, xenial)] config.test_manifest.ManifestConfigPrecedenceManual
17:28:26 | SUCCESS  [remote][(xenial, xenial)] config.test_manifest.ManifestConfigPrecedenceManual scenario has passed.
17:28:35 | INFO     Starting scenario (21/35): [remote][(xenial, xenial)] config.test_selection.TestSelectionDefault
17:28:43 | SUCCESS  [remote][(xenial, xenial)] config.test_selection.TestSelectionDefault scenario has passed.
17:28:54 | INFO     Starting scenario (22/35): [remote][(xenial, xenial)] config.test_selection.TestSelectionForced
17:29:00 | SUCCESS  [remote][(xenial, xenial)] config.test_selection.TestSelectionForced scenario has passed.
17:29:09 | INFO     Starting scenario (23/35): [remote][(xenial, xenial)] config.test_selection.TestSelectionExcludedJob
17:29:24 | SUCCESS  [remote][(xenial, xenial)] config.test_selection.TestSelectionExcludedJob scenario has passed.
17:29:34 | INFO     Starting scenario (24/35): [remote][(xenial, xenial)] config.test_selection.RemoteTestSelectionResolution
17:29:49 | SUCCESS  [remote][(xenial, xenial)] config.test_selection.RemoteTestSelectionResolution scenario has passed.
17:30:00 | INFO     Starting scenario (25/35): [remote][(xenial, xenial)] config.test_selection.TestPlanSelectionSkip
17:30:06 | SUCCESS  [remote][(xenial, xenial)] config.test_selection.TestPlanSelectionSkip scenario has passed.
17:30:17 | INFO     Starting scenario (26/35): [remote][(xenial, xenial)] config.test_selection.TestPlanPreselected
17:30:23 | SUCCESS  [remote][(xenial, xenial)] config.test_selection.TestPlanPreselected scenario has passed.
17:30:34 | INFO     Starting scenario (27/35): [remote][(xenial, xenial)] config.test_selection.TestPlanSelectionPreselectFailWrongName
17:30:40 | SUCCESS  [remote][(xenial, xenial)] config.test_selection.TestPlanSelectionPreselectFailWrongName scenario has passed.
17:30:49 | INFO     Starting scenario (28/35): [remote][(xenial, xenial)] config.test_selection.TestPlanSelectionPreselectNothing
17:30:57 | SUCCESS  [remote][(xenial, xenial)] config.test_selection.TestPlanSelectionPreselectNothing scenario has passed.
17:31:08 | INFO     Starting scenario (29/35): [remote][(xenial, xenial)] config.test_selection.TestPlanSelectionFilterEmpty
17:31:14 | SUCCESS  [remote][(xenial, xenial)] config.test_selection.TestPlanSelectionFilterEmpty scenario has passed.
17:31:27 | INFO     Starting scenario (30/35): [remote][(xenial, xenial)] config.test_selection.TestPlanSelectionFilter
17:31:27 | SUCCESS  [remote][(xenial, xenial)] config.test_selection.TestPlanSelectionFilter scenario has passed.
17:31:39 | INFO     Starting scenario (31/35): [remote][(xenial, xenial)] config.transport.TransportSecureIDSetInLauncherOnly
17:32:01 | SUCCESS  [remote][(xenial, xenial)] config.transport.TransportSecureIDSetInLauncherOnly scenario has passed.
17:32:13 | INFO     Starting scenario (32/35): [remote][(xenial, xenial)] config.transport.TransportSecureIDSetInConfigOnly
17:32:35 | SUCCESS  [remote][(xenial, xenial)] config.transport.TransportSecureIDSetInConfigOnly scenario has passed.
17:32:47 | INFO     Starting scenario (33/35): [remote][(xenial, xenial)] config.transport.TransportSecureIDOverwrittenByLauncher
17:33:04 | SUCCESS  [remote][(xenial, xenial)] config.transport.TransportSecureIDOverwrittenByLauncher scenario has passed.
17:33:16 | INFO     Starting scenario (34/35): [remote][(xenial, xenial)] desktop_env.launcher.GlxGears
17:33:34 | SUCCESS  [remote][(xenial, xenial)] desktop_env.launcher.GlxGears scenario has passed.
17:33:45 | INFO     Starting scenario (35/35): [remote][(xenial, xenial)] restart.launcher.Reboot
17:34:33 | SUCCESS  [remote][(xenial, xenial)] restart.launcher.Reboot scenario has passed.
```

Local Tests:
```
17:40:03 | INFO     Starting scenario (1/39): [local][('xenial',)] basic.return_codes.ReturnCodeIsOneOnFail
17:40:13 | SUCCESS  [local][('xenial',)] basic.return_codes.ReturnCodeIsOneOnFail scenario has passed.
17:40:17 | INFO     Starting scenario (2/39): [local][('xenial',)] basic.run-invocation.RunFailingTestplan
17:40:27 | SUCCESS  [local][('xenial',)] basic.run-invocation.RunFailingTestplan scenario has passed.
17:40:31 | INFO     Starting scenario (3/39): [local][('xenial',)] basic.run-invocation.RunTestplanWithEnvvar
17:40:41 | SUCCESS  [local][('xenial',)] basic.run-invocation.RunTestplanWithEnvvar scenario has passed.
17:40:45 | INFO     Starting scenario (4/39): [local][('xenial',)] basic.run-invocation.RunManualplan
17:40:55 | SUCCESS  [local][('xenial',)] basic.run-invocation.RunManualplan scenario has passed.
17:40:59 | INFO     Starting scenario (5/39): [local][('xenial',)] cert_blocker_comment.launcher.ManualJobFailed
17:41:09 | SUCCESS  [local][('xenial',)] cert_blocker_comment.launcher.ManualJobFailed scenario has passed.
17:41:14 | INFO     Starting scenario (6/39): [local][('xenial',)] cert_blocker_comment.launcher.ManualJobSkipped
17:41:25 | SUCCESS  [local][('xenial',)] cert_blocker_comment.launcher.ManualJobSkipped scenario has passed.
17:41:30 | INFO     Starting scenario (7/39): [local][('xenial',)] cert_blocker_comment.launcher.UserInteractVerifyJobFailed
17:41:40 | SUCCESS  [local][('xenial',)] cert_blocker_comment.launcher.UserInteractVerifyJobFailed scenario has passed.
17:41:44 | INFO     Starting scenario (8/39): [local][('xenial',)] cert_blocker_comment.launcher.UserInteractVerifyJobSkippedAfterRun
17:41:54 | SUCCESS  [local][('xenial',)] cert_blocker_comment.launcher.UserInteractVerifyJobSkippedAfterRun scenario has passed.
17:41:59 | INFO     Starting scenario (9/39): [local][('xenial',)] cert_blocker_comment.launcher.UserInteractVerifyJobSkippedBeforeRun
17:42:09 | SUCCESS  [local][('xenial',)] cert_blocker_comment.launcher.UserInteractVerifyJobSkippedBeforeRun scenario has passed.
17:42:13 | INFO     Starting scenario (10/39): [local][('xenial',)] cert_blocker_comment.launcher.UserInteractJobSkippedBeforeRun
17:42:23 | SUCCESS  [local][('xenial',)] cert_blocker_comment.launcher.UserInteractJobSkippedBeforeRun scenario has passed.
17:42:27 | INFO     Starting scenario (11/39): [local][('xenial',)] cert_blocker_comment.launcher.ManualJobSkippedWhenResumingSession
17:42:42 | SUCCESS  [local][('xenial',)] cert_blocker_comment.launcher.ManualJobSkippedWhenResumingSession scenario has passed.
17:42:46 | INFO     Starting scenario (12/39): [local][('xenial',)] config.chains.ConfigChainsPriority
17:42:55 | SUCCESS  [local][('xenial',)] config.chains.ConfigChainsPriority scenario has passed.
17:42:59 | INFO     Starting scenario (13/39): [local][('xenial',)] config.environment.CheckboxConfEnvvarCaseSensitive
17:43:09 | SUCCESS  [local][('xenial',)] config.environment.CheckboxConfEnvvarCaseSensitive scenario has passed.
17:43:13 | INFO     Starting scenario (14/39): [local][('xenial',)] config.environment.CheckboxConfXDG
17:43:23 | SUCCESS  [local][('xenial',)] config.environment.CheckboxConfXDG scenario has passed.
17:43:27 | INFO     Starting scenario (15/39): [local][('xenial',)] config.environment.CheckboxConfLocalHome
17:43:39 | SUCCESS  [local][('xenial',)] config.environment.CheckboxConfLocalHome scenario has passed.
17:43:43 | INFO     Starting scenario (16/39): [local][('xenial',)] config.environment.CheckboxConfSnap
17:43:54 | SUCCESS  [local][('xenial',)] config.environment.CheckboxConfSnap scenario has passed.
17:43:58 | INFO     Starting scenario (17/39): [local][('xenial',)] config.environment.CheckboxConfLauncher
17:44:08 | SUCCESS  [local][('xenial',)] config.environment.CheckboxConfLauncher scenario has passed.
17:44:12 | INFO     Starting scenario (18/39): [local][('xenial',)] config.environment.CheckboxConfLocalHomePrecedence
17:44:23 | SUCCESS  [local][('xenial',)] config.environment.CheckboxConfLocalHomePrecedence scenario has passed.
17:44:27 | INFO     Starting scenario (19/39): [local][('xenial',)] config.environment.CheckboxConfLocalResolutionOrder
17:44:39 | SUCCESS  [local][('xenial',)] config.environment.CheckboxConfLocalResolutionOrder scenario has passed.
17:44:44 | INFO     Starting scenario (20/39): [local][('xenial',)] config.exporters.ExporterSettingOptions
17:44:53 | SUCCESS  [local][('xenial',)] config.exporters.ExporterSettingOptions scenario has passed.
17:44:57 | INFO     Starting scenario (21/39): [local][('xenial',)] config.test_manifest.ManifestLauncherAuto
17:45:11 | SUCCESS  [local][('xenial',)] config.test_manifest.ManifestLauncherAuto scenario has passed.
17:45:15 | INFO     Starting scenario (22/39): [local][('xenial',)] config.test_manifest.ManifestLauncherManual
17:45:26 | SUCCESS  [local][('xenial',)] config.test_manifest.ManifestLauncherManual scenario has passed.
17:45:30 | INFO     Starting scenario (23/39): [local][('xenial',)] config.test_manifest.ManifestConfigCacheAuto
17:45:44 | SUCCESS  [local][('xenial',)] config.test_manifest.ManifestConfigCacheAuto scenario has passed.
17:45:48 | INFO     Starting scenario (24/39): [local][('xenial',)] config.test_manifest.ManifestConfigCacheManual
17:46:00 | SUCCESS  [local][('xenial',)] config.test_manifest.ManifestConfigCacheManual scenario has passed.
17:46:04 | INFO     Starting scenario (25/39): [local][('xenial',)] config.test_manifest.ManifestConfigPrecedenceAuto
17:46:18 | SUCCESS  [local][('xenial',)] config.test_manifest.ManifestConfigPrecedenceAuto scenario has passed.
17:46:22 | INFO     Starting scenario (26/39): [local][('xenial',)] config.test_manifest.ManifestConfigPrecedenceManual
17:46:34 | SUCCESS  [local][('xenial',)] config.test_manifest.ManifestConfigPrecedenceManual scenario has passed.
17:46:38 | INFO     Starting scenario (27/39): [local][('xenial',)] config.test_selection.TestSelectionDefault
17:46:48 | SUCCESS  [local][('xenial',)] config.test_selection.TestSelectionDefault scenario has passed.
17:46:53 | INFO     Starting scenario (28/39): [local][('xenial',)] config.test_selection.TestSelectionForced
17:47:01 | SUCCESS  [local][('xenial',)] config.test_selection.TestSelectionForced scenario has passed.
17:47:05 | INFO     Starting scenario (29/39): [local][('xenial',)] config.test_selection.TestSelectionExcludedJob
17:47:16 | SUCCESS  [local][('xenial',)] config.test_selection.TestSelectionExcludedJob scenario has passed.
17:47:20 | INFO     Starting scenario (30/39): [local][('xenial',)] config.test_selection.LocalTestSelectionResolution
17:47:30 | SUCCESS  [local][('xenial',)] config.test_selection.LocalTestSelectionResolution scenario has passed.
17:47:34 | INFO     Starting scenario (31/39): [local][('xenial',)] config.test_selection.TestPlanSelectionSkip
17:47:45 | SUCCESS  [local][('xenial',)] config.test_selection.TestPlanSelectionSkip scenario has passed.
17:47:49 | INFO     Starting scenario (32/39): [local][('xenial',)] config.test_selection.TestPlanPreselected
17:47:59 | SUCCESS  [local][('xenial',)] config.test_selection.TestPlanPreselected scenario has passed.
17:48:03 | INFO     Starting scenario (33/39): [local][('xenial',)] config.test_selection.TestPlanSelectionPreselectFailWrongName
17:48:14 | SUCCESS  [local][('xenial',)] config.test_selection.TestPlanSelectionPreselectFailWrongName scenario has passed.
17:48:18 | INFO     Starting scenario (34/39): [local][('xenial',)] config.test_selection.TestPlanSelectionPreselectNothing
17:48:29 | SUCCESS  [local][('xenial',)] config.test_selection.TestPlanSelectionPreselectNothing scenario has passed.
17:48:34 | INFO     Starting scenario (35/39): [local][('xenial',)] config.test_selection.TestPlanSelectionFilterEmpty
17:48:45 | SUCCESS  [local][('xenial',)] config.test_selection.TestPlanSelectionFilterEmpty scenario has passed.
17:48:49 | INFO     Starting scenario (36/39): [local][('xenial',)] config.test_selection.TestPlanSelectionFilter
17:48:50 | SUCCESS  [local][('xenial',)] config.test_selection.TestPlanSelectionFilter scenario has passed.
17:48:54 | INFO     Starting scenario (37/39): [local][('xenial',)] desktop_env.launcher.GlxGears
17:49:11 | SUCCESS  [local][('xenial',)] desktop_env.launcher.GlxGears scenario has passed.
17:49:16 | INFO     Starting scenario (38/39): [local][('xenial',)] installation.validation.DependencyInstallation
17:49:26 | SUCCESS  [local][('xenial',)] installation.validation.DependencyInstallation scenario has passed.
17:49:30 | INFO     Starting scenario (39/39): [local][('xenial',)] urwid.testplan.UrwidTestPlanSelection
17:49:54 | SUCCESS  [local][('xenial',)] urwid.testplan.UrwidTestPlanSelection scenario has passed.
```

**Review by kissiel — APPROVED** [review]
Nice! +1
I really hope this solves the s16 quest.

**Pull request merged**